javascript/integration-tests/AddOffer.spec.ts

```diff
@@ -0,0 +1,163 @@
+
+import { POINT_01_CTC } from '../src/constants';
+import { registerAddressAsync, RegisteredAddress } from '../src/examples/register-address';
+import { randomEthAddress } from '../src/utils';
+
+describe('AddOffer', (): void => {
+  let api: ApiPromise;
+  let borrower, lender: KeyringPair;
+  let loanTerms: PalletCreditcoinLoanTerms;
+  let borrowerRegAddr, lenderRegAddr: RegisteredAddress;
+  let askGuid, bidGuid: string;
+  let askOrderId: PalletCreditcoinAskOrderId;
+  let bidOrderId: PalletCreditcoinBidOrderId;
+
+  const blockchain = 'Ethereum';
+  const expirationBlock = 5;
+
+  beforeEach(async () => {
+    process.env.NODE_ENV = 'test';
+
+    const provider = new WsProvider('ws://127.0.0.1:9944');
+
+    api = await ApiPromise.create({ provider });
+
+    const keyring = new Keyring({ type: 'sr25519' });
+
+    lender = keyring.addFromUri('//Alice', { name: 'Alice' });
+    borrower = keyring.addFromUri('//Bob', { name: 'Bob' });
+    const lenderAddress = randomEthAddress();
+    const borrowerAddress = randomEthAddress();
+
+    loanTerms = api.createType<PalletCreditcoinLoanTerms>('PalletCreditcoinLoanTerms', {
+      amount: 1_000,
+      interestRate: 100,
+      maturity: 10
+    });
+
+    // register address for lender
+    let addr = await registerAddressAsync(api, lenderAddress, blockchain, lender);
+    expect(addr).toBeTruthy();
+
+    if (addr) {
+      lenderRegAddr = addr;
+      expect(lenderRegAddr.addressId).toBeTruthy();
+      askGuid = Guid.newGuid().toString();
+    } else {
+      throw new Error("Lender address wasn't registered successfully");
+    }
+
+    // register address for borrower
+    addr = await registerAddressAsync(api, borrowerAddress, blockchain, borrower);
+    expect(addr).toBeTruthy();
+
+    if (addr) {
+      borrowerRegAddr = addr;
+      expect(borrowerRegAddr.addressId).toBeTruthy();
+      bidGuid = Guid.newGuid().toString();
+    } else {
+      throw new Error("Borrower address wasn't registered successfully");
+    }
+
+    // AddAskOrder
+    let unsubscribe = await api.tx.creditcoin
+      .addAskOrder(lenderRegAddr.addressId, loanTerms, expirationBlock, askGuid)
+      .signAndSend(lender, { nonce: -1 }, async ({ dispatchError, events, status }) => {
+        expect(dispatchError).toBeFalsy();
+
+        if (status.isInBlock) {
+console.log(`====== DEBUG, AddAskOrder, events=${events}`);
+          const askOrderAdded = events.find(({ event: { method, section } }) => {
+            return section === 'creditcoin' && method === 'AskOrderAdded';
+          });
+
+          expect(askOrderAdded).toBeTruthy();
+
+          if (askOrderAdded) {
+              askOrderId = askOrderAdded.event.data[0] as PalletCreditcoinAskOrderId;
+              expect(askOrderId).toBeTruthy();
+
+              unsubscribe();
+          } else {
+              new Error("askOrderId wasn't found");
+          }
+        }
+      });
```

Comment:
@jfrank-summit, @nathanwhit is there an easy way to just wait for preconditions to be included in a block before continuing with the test and possibly avoid the subscription callbacks ? 

I was thinking we can simplify this whole code block by waiting for AddAskOrder to be included in a block and returning txHash. But then how do I get the askOrderId?